### PR TITLE
Implement Logging in the form of a Journal

### LINKED ISSUE #140: Implement a logging system to output results of virtual dice rolls
**User Story**
As a game designer, I'd like to communicate to the player the calculations involved in determining the success of their combat actions so that they may better understand why success succeeded and failures failed.

**Acceptance Criteria**

- Virtual dice rolls are logged internally
- This log is viewable by the player in the UI

**Notes**

Blocked by #138 and/or #136 

---

**Why is this feature needed? Please describe the problem your requested feature wants to solve**
The player has no knowledge of the virtual dice rolls that determine the success/failure of combat.

**Describe the solution you'd like**
A visual element should be added to the UI so that the player can access to view a log of all virtual dice rolls.

### PR DESCRIPTION
**Pull request by Pyxxil**
GitHub Issue (If applicable): #140

## PR Type

What kind of change does this PR introduce?
- Feature

## What is the current behavior?
Currently the player is provided a notification for the values they hit.

## What is the new behavior?
The player is now given a journal, which contains information such as:
 - The attack check to tell them if they hit the monster
 - Whether they hit the monster or not, and with how much damage
 - Healing (?)

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Tested code with current upstream repo
- [ ] Added new tests to the test suite for the feature
- [ ] Tested the full test suite, and ensured that the feature does not break current build.
- [x] Docs have been added/updated which fit (for bug fixes / features)
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal)
- [ ] Checked licensing of Frameworks/Libraries (if applicable) 

## Other information
I would like to have some changes which I'm not sure I can provide (regarding stylistic decisions and implementations, as well as if we want the logbook to also be in third person and contain thei players name).

Internal Issue (If applicable): #closes 140

### CODE REVIEW

File: src/features/journal/reducer.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+                'You rolled a ' +
+                    payload.ability +
+                    ' check  and got ' +
+                    payload.roll
+            );
+
+            if (payload.roll >= payload.check) {
+                newState.entries.push('The ability check succeeded!');
+            } else {
+                newState.push('The ability check failed');
```
Comment (Matteas-Eden) [inline src/features/journal/reducer.js]: Possible typo? `newState.push()` produces an error.

Also, would it be possible to add some more information to the journal? At the moment, a player knows what was rolled, but they don't know what they value they were rolling *against*. This means that they won't know why they succeeded or failed with the current implementation.

File: src/features/game-menus/index.js [hunk 2]
```diff
@@ -45,6 +46,7 @@ const GameMenus = ({ appState, dialog }) => {
                 <div className="flex-column">
                     <GameMusic sideMenu={sideMenu} />
                     <GameSettings />
+                    <GameJournal />
```
Comment (Matteas-Eden) [inline src/features/game-menus/index.js]: This button doesn't show up my on laptop. It gets cut off at the bottom of the screen, and there isn't any scrollbar so I wouldn't even know it was there if I didn't know better.

**Environment**
Version: 62b3aee42cd5d3aac3b3d9d66497b4dd030359df
OS: Arch Linux
Browser: Firefox v75

**Review by Matteas-Eden — CHANGES_REQUESTED** [review]
It's quite impressive and well made, though there are a couple changes I'm requesting.

In regards to how the journal should look, I'm a bit hesitant on the idea of adding another window. I feel like if the journal was implemented as a dialog, so that it overlaid on the top of the main screen and paused the game, that would look better. It's not a request at the moment though - just an idea.

File: src/features/journal/reducer.js (hunk 1 above)
Comment (Pyxxil) [reply src/features/journal/reducer.js]: I had added that to the payload, but didn't put it into the output as I wasn't sure how that should be worded.

File: src/features/game-menus/index.js (hunk 2 above)
Comment (Pyxxil) [reply src/features/game-menus/index.js]: I can't reproduce this on my laptop, so you'll have to do more investigation on your side.

**Review by KimberleyEvans-Parker — DISMISSED** [review]
Are you able to add a hotkey to the logger (L to open/close the logger)
I think it would also be helpful if there was something that seperated each item in the leg, e.g. each action was a bullet point or the first line for each action was bolded just so it is easier to see

**Comment by Matteas-Eden** [comment]
> 
> 
> Are you able to add a hotkey to the logger (L to open/close the logger)
> I think it would also be helpful if there was something that seperated each item in the leg, e.g. each action was a bullet point or the first line for each action was bolded just so it is easier to see

I second this, but I'd like to suggest `J` for *Journal* as the hotkey.

**Review by Matteas-Eden — APPROVED** [review]
Looks great, well done @Pyxxil.

**Pull request merged**